frontend/src/tests/application-tests.js

```diff
@@ -1,19 +1,110 @@
     describe("Public route tests", () => {
+        beforeAll(async () => {
+            await databaseSeeder.seed({ apiGET, apiPOST });
+            session = databaseSeeder.seededData.session;
+            position = databaseSeeder.seededData.position;
+            resp = await apiGET(`/debug/active_user`);
+            expect(resp).toHaveStatus("success");
+            admin = {
+                utorid: resp.payload.utorid,
+                roles: resp.payload.roles,
+            };
+        });
+
         it.todo("Get survey.js posting data through public route");
         it.todo(
             "Survey.js posting data is pre-filled based on prior applicant/application data"
         );
-        it.todo("Can submit survey.js data via the public postings route");
+
+        it("Can submit survey.js data via the public postings route", async () => {
+            //Create a new posting
+            resp = await apiPOST(
+                `/admin/sessions/${session.id}/postings`,
+                postingData
+            );
+            expect(resp).toHaveStatus("success");
+            Object.assign(posting, resp.payload);
+            checkPropTypes(postingPropTypes, posting);
+            expect(posting.id).not.toBeNull();
+
+            //Set position for posting
+            resp = await apiPOST(
+                `/admin/postings/${posting.id}/posting_positions`,
+                { hours: 20, num_positions: 2, position_id: position.id }
+            );
+            expect(resp).toHaveStatus("success");
+
+            //Create and switch to a ta only user
+            resp = await apiPOST("/debug/users", taOnlyUser);
+            expect(resp).toHaveStatus("success");
+
+            //Create and submit survey.js data
+            surveyData = {
+                answers: {
+                    utorid: "matthewc",
+                    student_number: "1000123456",
+                    first_name: "Matthew",
+                    last_name: "Cambell",
+                    email: "test@test.ca",
+                    phone: "6471234567",
+                    program: "M",
+                    program_start: "2017-09-05",
+                    department: "cs",
+                    previous_university_ta: true,
+                    some_entry: false,
+                    previous_department_ta: true,
+                    previous_other_university_ta: false,
+                    previous_experience_summary: "some previous experience",
+                    position_preferences: {
+                        [position.position_code]: 3,
+                    },
+                    comments: "some additional comments",
+                },
+            };
+
+            //Submit survey.js data
+            resp = await apiPOST(
+                `/public/postings/${posting.url_token}/submit`,
+                surveyData,
+                true
+            );
+            expect(resp).toHaveStatus("success");
```

Comment:
Do we want to also `expect(resp.payload).toStrictEqual()` the expected payload?

Reply:
According to Jason, verification like so at this stage doesn't confirm anything meaningful, and since application data is verified in another test case, I'm  just confirming success here.